proposals/0017-conforming-literals.md

```diff
@@ -82,6 +82,100 @@ adopting Clang).
+the impact of this change on existing shaders used in production software.
+
+Tests were performed by both Microsoft and Google using private test suites. In
+Google's testing, *all tests passed all tests with no issues*.
+
+#### Microsoft Test Suite
+
+The test cases are PIX captures running on Xbox Series X. Most of the captures
+are rendered frames of in-development or shipped games, or sample content
+produced from external developers working on games or engine technology. The
+remaining captures are test programs developed internally at Microsoft.
+
+For the test runs the image difference tolerance was zero, meaning that any
+frame which had even a single bit difference in output was marked as a failure.
+
+#### Summary of Results
+
+Of 271 capture test cases, 10 shaders failed to compile with the new compiler
+and 19 capture tests produced a binary different frame rendering. The
+binary-different results can be further broken down by severity:
+
+* Extremely minor - 8 test cases
+* Minor - 3 test cases
+* Subtle visible differences - 2 test cases
+* Catastrophic visible differences - 6 test cases
+
+##### Compile Failures
+
+All 10 shaders that failed to compile hit a variation of the issue reported as
+[DXC/#6315](https://github.com/microsoft/DirectXShaderCompiler/issues/6315). The
+failing shaders call `asfloat16` with a floating literal value, but `asfloat16`
+does not have a full set of overloads.
+
+##### Extremely Minor
+
+Extremely minor failures produced image differences so minor that they are not
+even visible with assistive image difference tools. They appear in binary
+differences as pixel values being only a few bits off (in most cases 1-2 bits
+per pixel).
+
+##### Minor
+
+Minor failures produced image differences that appeared visible with assistive
+image difference tools, but not visibly distinct in final rendered images. These
+appeared as pixels that deviate from the reference value by more than a few
+bits. Examples of the image diff view are:
+
+![Minor Differences Example 1](0017-assets/DiffExample1.png)
+![Minor Differences Example 2](0017-assets/DiffExample2.png)
+
+##### Subtle Visible Differences
+
+A small number of tests had significant differences in clusters of pixels large
+enough to produce minor visual artifacts. These differences are significant
+enough and represent inaccuracies in rendered output that we would expect a
+software developer to be concerned by and wish to address even though the
+overall image still rendered mostly correctly.
+
+##### Catastrophic Visible Differences
+
+We have 6 test cases, all from the same software title which had severe
+obviously incorrect rendering which would make the title unplayable. One
+important note about this title is that it is an Xbox & Windows exclusive title.
+
+#### Conclusions Drawn
+
+The testing here holds with the core thesis that developer impact will be
+minimal and that most shaders are resilient to the changes in precision. Further

```

Comment:
> developer impact will be minimal

I think I would say this slightly differently. That the number of titles impacted should be minimal, but the impact to a single developer could require them to rewrite some shaders. 

What would the fix be for the catastrophic failures? Would it be to use the previous version of hlsl with the `-HV` flag? Or use a suffix on some of the constants?

Reply:
Possibly, or to find out the cause of the issue since.  Easy to miss from the PR, but this is in the context of a proposal for a change in HLSL 202x, so DXC behavior for other language versions won't change.

Reply:
Recall that the current DXC behavior doesn't match the documentation: https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-appendix-grammar#floating-point-numbers:

> Use the “L” suffix to specify a full 64-bit precision floating-point literal. A 32-bit float literal is the default.
>
> For example, the compiler recognizes the following literal value as a 32-bit precision floating-point literal and ignores the lower bits:
>
>
> `double x = -0.6473313946860445;`
> The compiler recognizes the following literal value as a 64-bit precision floating-point literal:
>
> `double x = -0.6473313946860445L;`

Reply:
@dmpots, are you okay with my updated wording here?

Reply:
Yes, that looks good. It clearly separates the expected impact across all titles vs. the work an individual title might have to do if impacted by the change. Thanks!

Reply:
I filed #184 to track the remaining outstanding question here about any mitigations we may want to take for the catastrophic failures.

Since @dmpots is okay with the updated wording I'll resolve this thread.